src/SessionSetup/RetrieveInstancesWidgetTest.cpp

```diff
@@ -80,4 +216,80 @@ TEST_F(RetrieveInstancesWidgetTest, FilterTextChanged) {
   EXPECT_EQ(arguments.at(0).toString(), "test text");
 }
 
+TEST_F(RetrieveInstancesWidgetTest, StartSuccessfulDefault) {
+  EXPECT_CALL(mock_retrieve_instances_,
+              LoadProjectsAndInstances(
+                  std::optional<Project>(std::nullopt),
+                  Client::InstanceListScope(Client::InstanceListScope::kOnlyOwnInstances)))
+      .WillOnce(Return(Future<ErrorMessageOr<RetrieveInstances::LoadProjectsAndInstancesResult>>{
+          kInitialTestDataDefault}));
+
+  widget_.Start();
+  QCoreApplication::processEvents();
+
+  EXPECT_EQ(loading_started_spy_.count(), 1);
+  EXPECT_EQ(loading_successful_spy_.count(), 1);
+  EXPECT_EQ(loading_failed_spy_.count(), 0);
+  EXPECT_EQ(initial_loading_failed_spy_.count(), 0);
```

Comment:
I wonder if verifying this actually add any signal (pun maybe intended). In the end, you're testing here if all UI elements are filled as expected, and that's what's important. If the Widget were to be refactored at some point to not use those signals, would we care as long as the result is the same?

Reply:
These signals are used as an interface to the whoever uses the widget (in our case ConnectToStadiaWidget), hence I would like to keep these tests. wdyt?

(InitialLoadingFailed is actually only used internally, but just removing that seems also inconsistent)

Reply:
Ah right, I didn't realize they were used for external communication as well, somehow I thought they were just used internally. Then I agree, keep the tests.